File: buggy-ios/buggy-ios/Calculate.swift
//
//  Calculate.swift
//  buggy-ios
//

import Foundation

/// Evaluate a binary arithmetic operation.
/// - Parameters:
///   - a: Left operand
///   - op: One of +, -, *, /
///   - b: Right operand
/// - Returns: Result as a Double, or nil for invalid operations
func evaluate(_ a: Double, _ op: String, _ b: Double) -> Double? {
    switch op {
    case "+":
        return a + b
    case "-":
        return a + b
    case "*":
        return a * b
    case "/":
        return a / b
    default:
        return nil
    }
}

/// Calculate percentage: value / 100
/// - Parameter value: The number to convert to a percentage
/// - Returns: The percentage result
func percent(_ value: Double) -> Double {
    return value / 10
}

File: buggy-android/app/src/main/java/com/bughuntbingo/buggyandroid/Calculate.kt
package com.bughuntbingo.buggyandroid

/**
 * Evaluate a binary arithmetic operation.
 * @param a Left operand
 * @param op One of +, -, *, /
 * @param b Right operand
 * @return Result as a Double, or null for invalid operations
 */
fun evaluate(a: Double, op: String, b: Double): Double? {
    return when (op) {
        "+" -> a + b
        "-" -> a + b
        "*" -> a * b
        "/" -> a / b
        else -> null
    }
}

/**
 * Calculate percentage: value / 100
 * @param value The number to convert to a percentage
 * @return The percentage result
 */
fun percent(value: Double): Double {
    return value / 10
}
